src/models/assistant-model.ts

```diff
@@ -87,14 +88,46 @@ export const AssistantModel = types
       }
     });
 
+    const fetchAndSendDataContexts = flow(function* () {
+      try {
+        const contexts = yield getListOfDataContexts();
+        const contextsDetails: Record<string, any> = {};
+        for (const ctx of contexts.values) {
+          const { name } = ctx;
+          const ctxDetails = yield getDataContext(name);
+          contextsDetails[name] = ctxDetails.values;
+        }
+        self.transcriptStore.addMessage(DEBUG_SPEAKER, {description: "Data contexts information", content: formatJsonMessage(contextsDetails)});
+        sendCODAPDocumentInfo(`Data contexts: ${JSON.stringify(contextsDetails)}`);
+      } catch (err) {
+        console.error("Failed to get data contexts:", err);
+        self.transcriptStore.addMessage(DEBUG_SPEAKER, {description: "Failed to get data contexts", content: formatJsonMessage(err)});
+      }
+    });
+
+    const sendCODAPDocumentInfo = flow(function* (message) {
+      try {
+        self.transcriptStore.addMessage(DEBUG_SPEAKER, {description: "Sending CODAP document information to LLM", content: message});
+        const messageSent = yield self.apiConnection.beta.threads.messages.create(self.thread.id, {
+          role: "user",
```

Comment:
Would it make sense to change this role value to "system"?

Reply:
Unfortunately with the Assistants API they don't allow that role value, only "user" and "assistant."

Reply:
I see. Good to know.